File: snappii/app/src/main/java/com/home/snappii/AppApplication.java
package com.home.snappii;


import android.app.Application;
import android.support.multidex.MultiDex;

import com.home.snappii.di.Component;
import com.home.snappii.di.DaggerComponent;
import com.home.snappii.di.MainModule;
import com.home.snappii.di.NetworkModule;

public class AppApplication extends Application {

    private Component component;

    @Override
    public void onCreate() {
        super.onCreate();

        MultiDex.install(this);

        component = DaggerComponent.builder()
                .networkModule(new NetworkModule(this))
                .mainModule(new MainModule(this))
                .build();
    }

    public Component getComponent() {
        return component;
    }
}

File: Trucks/app/src/main/java/ru/rsprm/AppApplication.kt
package ru.rsprm

import android.app.Application
import ru.rsprm.di.AppComponent
import ru.rsprm.di.DaggerAppComponent
import ru.rsprm.di.Module
import ru.rsprm.di.NetworkModule

class AppApplication : Application() {

    private var component: AppComponent? = null

    override fun onCreate() {
        super.onCreate()
        component = initDagger()
    }

    fun getComponent(): AppComponent {
        return component!!
    }

    private fun initDagger(): AppComponent {
        return DaggerAppComponent
            .builder()
            .networkModule(NetworkModule(this, getString(R.string.url)))
            .module(Module(this))
            .build()
    }
}
